### DISCUSSION TITLE
Lossless FLIF encoding

### DISCUSSION THREAD

**Discussion by n00mkrad** [dlemstra/Magick.NET#780] [Help]
Hello, is there a lossless mode for FLIF, like there is for WEBP etc.?

**Comment by dlemstra** [comment]
https://github.com/FLIF-hub/FLIF:
>FLIF is a lossless image format based on MANIAC compression. MANIAC (Meta-Adaptive Near-zero Integer Arithmetic Coding) is a variant of CABAC (context-adaptive binary arithmetic coding), where the contexts are nodes of decision trees which are dynamically learned at encode time.

Not sure what you ar aasking?

**Comment by n00mkrad** [comment]
FLIF can be lossless or lossy, just like WEBP.

By default, it's lossy in Magick.NET, and the quality can be adjusted with the MagickImage.Quality property. This can be proved by encoding FLIF with quality 0, the image will have a tiny file size  and noticably worse quality.

Now I want to know if a separate define is needed like for WEBP (`img.Settings.SetDefine(MagickFormat.WebP, "lossless", true);`, or if quality 100 means lossless.

**Comment by dlemstra** [comment]
You don't need a separate define. Setting the quality to 100 (the default value) means lossless. Did not know that FLIF also supported lossy encoding.
